go/apps/linux_host/linux_host.go

```diff
@@ -17,247 +17,534 @@ package main
+		if parent == nil {
+			usage("No host tao available, verify -parent_type or $%s\n", tao.HostChannelTypeEnvVar)
+		}
+		host, err = tao.NewStackedLinuxHost(hostPath(), domain.Guard, tao.ParentFromConfig(tc), childFactory)
+	}
+	failIf(err, "Can't create host")
+
+	return host
+}
+
+func initHost(domain *tao.Domain) {
+	var cfg tao.LinuxHostConfig
+
+	configureFromOptions(&cfg)
+	_ = loadHost(domain, &cfg)
+
+	// If we get here, keys were created and flags must be ok.
+
+	file, err := util.CreatePath(hostConfigPath(), 0777, 0666)
+	failIf(err, "Can't create host configuration")
+	cs := proto.MarshalTextString(&cfg)
+	fmt.Fprint(file, cs)
+	file.Close()
+}
+
+func showHost(domain *tao.Domain) {
+	cfg := configureFromFile()
+	configureFromOptions(cfg)
+	host := loadHost(domain, cfg)
+	fmt.Printf("%v\n", host.HostName())
+}
+
+func isBoolFlagSet(name string) bool {
+	f := flag.Lookup("logtostderr")
+	if f == nil {
+		return false
+	}
+	v, ok := f.Value.(flag.Getter).Get().(bool)
+	return ok && v
+}
+
+func daemonize() {
+	// For our purposes, "daemon" means being a session leader.
+	sid, _, errno := syscall.Syscall(syscall.SYS_GETSID, 0, 0, 0)
+	var err error
+	if errno != 0 {
+		err = errno
+	}
+	failIf(err, "Can't get process SID")
+	if int(sid) != syscall.Getpid() {
+		if syscall.Getpid() == syscall.Getpid() {
+			fmt.Fprintf(noise, "Forking to enable setsid\n")
+		} else {
+			fmt.Fprintf(noise, "Forking anyway\n")
+		}
+		// No daemonize, but we can just fork/exec and exit
+		path, err := os.Readlink("/proc/self/exe")
+		failIf(err, "Can't get path to self executable")
+		// special case: keep stderr if -logtostderr or -alsologtostderr
+		stderr := os.Stderr
+		if !isBoolFlagSet("logtostderr") && !isBoolFlagSet("alsologtostderr") {
+			stderr = nil
+		}
+		spa := &syscall.SysProcAttr{
+			Setsid: true, // Create session.
+			// Setpgid: true, // Set process group ID to new pid (SYSV setpgrp)
+			// Setctty: true, // Set controlling terminal to fd Ctty (only meaningful if Setsid is set)
+			// Noctty: true, // Detach fd 0 from controlling terminal
+			// Ctty: 0, // Controlling TTY fd (Linux only)
+		}
+		daemon := exec.Cmd{
+			Path:        path,
+			Args:        os.Args,
+			Stderr:      stderr,
+			SysProcAttr: spa,
+		}
+		err = daemon.Start()
+		failIf(err, "Can't become daemon")
+		fmt.Fprintf(noise, "Linux Tao Host running as daemon\n")
+		os.Exit(0)
+	} else {
+		fmt.Fprintf(noise, "Already a session leader?\n")
+	}
+}
+
+func startHost(domain *tao.Domain) {
+
+	if *options.Bool["daemon"] && *options.Bool["foreground"] {
+		usage("Can supply only one of -daemon and -foreground")
+	}
+	if *options.Bool["daemon"] {
+		daemonize()
+	}
+
+	cfg := configureFromFile()
+	configureFromOptions(cfg)
+	host := loadHost(domain, cfg)
+
+	sockPath := path.Join(hostPath(), "admin_socket")
+	// Make sure callers can read the admin socket directory
+	err := os.Chmod(path.Dir(sockPath), 0755)
+	failIf(err, "Can't change permissions")
```

Comment:
Please add information in this function about which files are unreadable/unresolvable etc.